exonum-java-binding-common/src/main/java/com/exonum/binding/common/proofs/list/CheckedListProofImpl.java

```diff
@@ -60,10 +59,15 @@ public HashCode getRootHash() {
   }
 
   @Override
-  public ProofStatus getStatus() {
+  public ListProofStatus getProofStatus() {
     return proofStatus;
   }
 
+  @Override
+  public boolean isValid() {
+    return proofStatus == ListProofStatus.VALID;
+  }
+
   private void checkValid() {
     checkState(proofStatus == ListProofStatus.VALID, "Proof is not valid: %s", proofStatus);
```

Comment:
We could avoid code duplication here `checkState(isValid(), "Proof is not valid: %s", proofStatus);`, what do you think?
Same in `CheckedFlatMapProof`

Reply:
Nice.